python_modules/dagster/dagster/core/execution/context/compute.py

```diff
@@ -237,6 +246,53 @@ def get_tag(self, key: str) -> Optional[str]:
         """
         return self._step_execution_context.get_tag(key)
 
+    def has_events(self) -> bool:
+        return bool(self._events)
+
+    def consume_events(self) -> Iterator[DagsterEvent]:
+        """Yields all user-generated events that have been recorded from this context since the last consumption event.
```

Comment:
By "last consumption event", do you mean the last time `consume_events` was called? I think the mixed uses of the word "event" are a little confusing. Also, a tiny bit inaccurate in the sense that, if `consume_events` has never been called, but events have been logged, it will return stuff.